ocaml/xapi/xapi_sr.ml

```diff
@@ -787,30 +787,38 @@ let scan ~__context ~sr =
   SRScanThrottle.execute (fun () ->
       transform_storage_exn (fun () ->
           let sr_uuid = Db.SR.get_uuid ~__context ~self:sr in
-          let vs =
-            C.SR.scan (Ref.string_of task)
-              (Storage_interface.Sr.of_string sr_uuid)
-          in
-          let db_vdis =
-            Db.VDI.get_records_where ~__context
-              ~expr:(Eq (Field "SR", Literal sr'))
-          in
-          update_vdis ~__context ~sr db_vdis vs ;
+
           let sr_info =
             C.SR.stat (Ref.string_of task)
               (Storage_interface.Sr.of_string sr_uuid)
           in
-          let virtual_allocation =
-            List.fold_left Int64.add 0L
-              (List.map (fun v -> v.Storage_interface.virtual_size) vs)
-          in
-          Db.SR.set_virtual_allocation ~__context ~self:sr
-            ~value:virtual_allocation ;
-          Db.SR.set_physical_size ~__context ~self:sr ~value:sr_info.total_space ;
-          Db.SR.set_physical_utilisation ~__context ~self:sr
-            ~value:(Int64.sub sr_info.total_space sr_info.free_space) ;
-          Db.SR.remove_from_other_config ~__context ~self:sr ~key:"dirty" ;
-          Db.SR.set_clustered ~__context ~self:sr ~value:sr_info.clustered
+          match sr_info with
+          | {health; _} when health <> Healthy ->
+              raise Storage_interface.(Storage_error (Sr_unhealthy health))
+          | _ ->
+              let vs =
```

Comment:
What does `v` stand for? Volume? So vs = volumes?

Reply:
v has the type `vdi_info`, so vs is the list of vdi_info inside the scanned sr